micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -250,4 +264,91 @@ private SessionFactory unwrap(EntityManagerFactory entityManagerFactory) {
         }
     }
 
+    /**
+     * Register hibernate loadEvent listener - event trigger the metrics registration,
+     * only select queries are recorded in QueryStatistics {@link org.hibernate.stat.spi.StatisticsImplementor}
+     * @param meterRegistry meterRegistry to use
+     */
+    private void registerListeners(MeterRegistry meterRegistry) {
+        if (Objects.nonNull(sessionFactory) && sessionFactory instanceof SessionFactoryImplementor) {
```

Comment:
nit: `instanceof` works as a null check so the `Objects.nonNull(sessionFactory)` is not necessary.